Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php

```diff
@@ -209,7 +209,7 @@ private function whenNodeAggregateWithNodeWasCreated(NodeAggregateWithNodeWasCre
         }
 
         $propertyValues = $event->initialPropertyValues->getPlainValues();
-        $uriPathSegment = $propertyValues['uriPathSegment'] ?? '';
+        $uriPathSegment = $propertyValues['uriPathSegment'] ?? $event->nodeAggregateId->value;
```

Comment:
we discussed an alternative in line 273 but it doesnt really work :D 

```php
                $uriPath = $this->nodeTypeManager->getNodeType($parentNode->getNodeTypeName())?->isOfType(NodeTypeNameFactory::NAME_SITE)
                    ? $uriPathSegment
                    : $parentNode->getUriPath() . '/' . $uriPathSegment;
```

Reply:
Still think resilience against missing uri path segments is better than adding an even tighter bond to the Neos NodeTypes. Even if this would work ;-)

The fallback way makes uriPathes more like a soft concern (which it is technically anyways) and allows routing of documents in cases where it is missing for any reason. 

Only thing that would speak against that would be a case where the node-identifier based fallback does harm anything but i do not see a way for this to happen.